controller/konnect/ops/ops_kongpluginbinding.go

```diff
@@ -66,34 +67,51 @@ func updatePlugin(
 	ctx context.Context,
 	sdk sdkops.PluginSDK,
 	cl client.Client,
-	pb *configurationv1alpha1.KongPluginBinding,
+	pluginBinding *configurationv1alpha1.KongPluginBinding,
 ) error {
-	controlPlaneID := pb.GetControlPlaneID()
+	controlPlaneID := pluginBinding.GetControlPlaneID()
 	if controlPlaneID == "" {
-		return CantPerformOperationWithoutControlPlaneIDError{Entity: pb, Op: UpdateOp}
+		return CantPerformOperationWithoutControlPlaneIDError{Entity: pluginBinding, Op: UpdateOp}
 	}
 
-	pluginInput, err := kongPluginBindingToSDKPluginInput(ctx, cl, pb)
+	pluginInput, err := kongPluginBindingToSDKPluginInput(ctx, cl, pluginBinding)
 	if err != nil {
 		return err
 	}
 
+	id := pluginBinding.GetKonnectID()
 	_, err = sdk.UpsertPlugin(ctx,
 		sdkkonnectops.UpsertPluginRequest{
 			ControlPlaneID: controlPlaneID,
-			PluginID:       pb.GetKonnectID(),
+			PluginID:       id,
 			Plugin:         *pluginInput,
 		},
 	)
 
-	// TODO: handle already exists
-	// Can't adopt it as it will cause conflicts between the controller
-	// that created that entity and already manages it, hm
-	if errWrap := wrapErrIfKonnectOpFailed[configurationv1alpha1.KongPluginBinding](err, UpdateOp, pb); errWrap != nil {
-		SetKonnectEntityProgrammedConditionFalse(pb, "FailedToUpdate", errWrap.Error())
+	if errWrap := wrapErrIfKonnectOpFailed(err, UpdateOp, pluginBinding); errWrap != nil {
+		// plugin update operation returns an SDKError instead of a NotFoundError.
+		var sdkError *sdkkonnecterrs.SDKError
+		if errors.As(errWrap, &sdkError) {
+			switch sdkError.StatusCode {
+			case 404:
+				logEntityNotFoundRecreating(ctx, pluginBinding, id)
+				if err := createPlugin(ctx, cl, sdk, pluginBinding); err != nil {
+					return FailedKonnectOpError[configurationv1alpha1.KongPluginBinding]{
+						Op:  UpdateOp,
+						Err: err,
+					}
+				}
+				return nil
+			default:
+				return FailedKonnectOpError[configurationv1alpha1.KongPluginBinding]{
+					Op:  UpdateOp,
+					Err: sdkError,
+				}
+			}
+		}
+
 		return errWrap
 	}
```

Comment:
I think that `createPlugin` step was here before but was removed intentionally as it is not really needed due to `UpsertPlugin` method we call here which does the same already logically.

Reply:
Yep, that's a good catch. I've reverted the change 👍